tests/tensorflow/test_tensorflow_estimator_export.py

```diff
@@ -0,0 +1,273 @@
+    mlflow_repo_path = os.getcwd()
+
+    with chdir(tmpdir):
+        # change current working directory to be a temporary directory,
+        # to prevent importing current repo mlflow module.
+        _execute_in_conda_env(
+            conda_env,
+            f"python {exec_py_path} {task_type} {save_as_type} "
+            f"{save_path if save_path else ''}",
+            install_mlflow=False,
+            command_env={"MLFLOW_TRACKING_URI": tracking_uri, "MLFLOW_REPO_PATH": mlflow_repo_path},
+        )
+        with open(output_data_file_path, "rb") as f:
+            inference_df, expected_results_df, run_id = pickle.load(f)
+            return (
+                ModelDataInfo(
+                    inference_df=inference_df,
+                    expected_results_df=expected_results_df,
+                    raw_results=None,
+                    raw_df=None,
+                    run_id=run_id,
+                ),
+                tracking_uri,
+            )
+
+
+def test_load_model_from_remote_uri_succeeds(tmpdir, model_path, mock_s3_bucket):
+    model_data_info, _ = save_or_log_tf_model_by_mlflow128(
+        str(tmpdir), "iris", "save_model", model_path
+    )
+
+    artifact_root = "s3://{bucket_name}".format(bucket_name=mock_s3_bucket)
+    artifact_path = "model"
+    artifact_repo = S3ArtifactRepository(artifact_root)
+    artifact_repo.log_artifacts(model_path, artifact_path=artifact_path)
+
+    model_uri = artifact_root + "/" + artifact_path
+    infer = mlflow.tensorflow.load_model(model_uri=model_uri)
+    feed_dict = {
+        df_column_name: tf.constant(model_data_info.inference_df[df_column_name])
+        for df_column_name in list(model_data_info.inference_df)
+    }
+    raw_preds = infer(**feed_dict)
+    pred_dict = {column_name: raw_preds[column_name].numpy() for column_name in raw_preds.keys()}
+    for col in pred_dict:
+        np.testing.assert_allclose(
+            np.array(pred_dict[col], dtype=np.float),
+            np.array(model_data_info.raw_results[col], dtype=np.float),
+        )
+
+
+def test_iris_model_can_be_loaded_and_evaluated_successfully(tmpdir, model_path):
+    model_data_info, _ = save_or_log_tf_model_by_mlflow128(
+        str(tmpdir), "iris", "save_model", model_path
+    )
+
+    def load_and_evaluate():
+
+        infer = mlflow.tensorflow.load_model(model_uri=model_path)
+        feed_dict = {
+            df_column_name: tf.constant(model_data_info.inference_df[df_column_name])
+            for df_column_name in list(model_data_info.inference_df)
+        }
+        raw_preds = infer(**feed_dict)
+        pred_dict = {
+            column_name: raw_preds[column_name].numpy() for column_name in raw_preds.keys()
+        }
+        for col in pred_dict:
+            np.testing.assert_array_equal(pred_dict[col], model_data_info.raw_results[col])
+
+    load_and_evaluate()
+
+    with tf.device("/CPU:0"):
+        load_and_evaluate()
```

Comment:
What is the purpose of this block?

Reply:
This comes from existing testing code, means forcing to use CPU-0 ?

Reply:
we can remove the with device thought

Reply:
Let's check the PR that added this block.